src/endpoint/s3/ops/s3_get_bucket.js

```diff
@@ -88,9 +88,11 @@ function cont_tok_to_key_marker(cont_tok) {
     }
 }
 
-function key_marker_to_cont_tok(key_marker) {
-    if (!key_marker) return;
-    const j = JSON.stringify({ key: key_marker });
+function key_marker_to_cont_tok(key_marker, objects_arr, is_truncated) {
+    if (!key_marker && !is_truncated) return;
+    // next marker is the key marker we got or the key of the last item in the objects list.
+    const next_marker = key_marker || (objects_arr && objects_arr[objects_arr.length - 1].key);
```

Comment:
`objects_arr &&` this will always be true because we return an empty array.
Maybe you can check `!_.isEmpty(objects_arr)`

Reply:
when having only folders and delimiter is passed - key marker exists and we won't get there.
when having only folders and delimiter is not passed - we will have the folder name and content (according to the max keys) in the objects list.

Reply:
If we aren't reaching that if with [] array then just remove the check since it is irrelevant.

Reply:
ok, deleted